### DISCUSSION TITLE
How to use environment variables to establish configuration

### DISCUSSION THREAD

**Discussion by jfoxworth** [vercel/next.js#24796] [Help]
I have a react app with next.js. It is designed to be run at different locations. I would like to use an environment variable to establish some configuration settings. For example, one component sets up a websocket for a video and the config file holds the URL for that websocket. There are multiple config settings such as this that vary depending upon location. Right now, there is a config file for each location and I have to manually change it depending upon the location.

I would like for an environment variable to determine which file is read. 

The problem is that something like this is not allowed : 

    import { config } from '../'+process.env.NEXT_PUBLIC_BUILD_LOC+'/config';

A dynamic import using the method below results in a promise which is not resolved in time - resulting in an error:

    const conf = import(`../public/Builds/${process.env.NEXT_PUBLIC_BUILD_LOC}/config.js`);

What is the recommended or proper method of doing this?

**Comment by pvlcor** [comment]
have you considered creating some meta-config above all the ones you have? smth like
```
import cfgLocation1 from '...'
import cfgLocation2 from '...'
import cfgLocation3 from '...'

const metaConfig = {
    cfgLocation1,
    cfgLocation2,
    cfgLocation3,
}

export default metaConfig[`cfg${process.env.NEXT_PUBLIC_BUILD_LOC}`]
```

**Reply by jfoxworth** [reply]
Yes. That's actually how I originally solved the issue. However, the higher ups want that data to come in from outside the app.

**Reply by pvlcor** [reply]
Interesting, what do you mean by "data to come in from outside the app"? 
Metaconfig as a standalone npm package can be, technically, outside the app ;)
